graphql/batch_scheduler.go

```diff
@@ -10,15 +12,33 @@ func NewImmediateGoroutineScheduler() WorkScheduler {
 	return &immediateGoroutineScheduler{}
 }
 
-type immediateGoroutineScheduler struct{}
+type immediateGoroutineScheduler struct {
+	wg    sync.WaitGroup
+	count int64
+}
 
 func (q *immediateGoroutineScheduler) Run(resolver UnitResolver, initialUnits ...*WorkUnit) {
 	r := &immediateGoroutineSchedulerRunner{}
 	r.runEnqueue(resolver, initialUnits...)
-
 	r.wg.Wait()
 }
 
+func (q *immediateGoroutineScheduler) RunAll(ctx context.Context, executionUnits []*ExecutionUnit) {
+	q.runEnqueue(executionUnits)
+	q.wg.Wait()
+}
+
+func (q *immediateGoroutineScheduler) runEnqueue(executionUnits []*ExecutionUnit) {
```

Comment:
This is a little confusing: there are two `runEnqueue` methods, one on the scheduler and one on the runner.  It seems like maybe the responsibilities between these two structs are blurring a little bit?

Reply:
Maybe we should rename the methods to `runEnqueueWorkUnits` and `runEnqueueExecutionUnits`, then move the latter to `immediateGoroutineSchedulerRunner`.